mindsdb/api/mongo/responders/find.py

```diff
@@ -40,7 +40,9 @@ def result(self, query, request_env, mindsdb_env, session):
 
             for key in where_data:
                 if key not in columns:
-                    raise Exception(f"Unknown column '{key}'. Only columns from this list can be used in query: {', '.join(columns)}")
+                    sub_key = key[:key.find('.')] if '.' in key else key
+                    if sub_key not in columns:
+                        raise Exception(f"Unknown column '{key}'. Only columns from this list can be used in query: {', '.join(columns)}")
```

Comment:
I think that for now it might be best if we just skip this bit for Mongo, it makes more sense for SQL, but mongo is unstructred anyway, people might well want to predict with data different from the training set, even if EOD the columns there will be ignored.

So unless you're strongly against (in which case let's discuss) I'd say drop the schema check for mongo entirely.